### PR TITLE
Process data models

### PR DESCRIPTION
**Pull request by fernando-silvabr66**
Create models with custom methods:

- Create model for Users, including a method that returns the 3 most recent posts for a given user.
- Create model for Posts, including a method that updates the posts counter for a user and a method that returns the 5 most recent comments for a given post.
- Create model for Comments, including a method that updates the comments counter for a post.
- Cresste model for Likes, including a method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def five_most_recent_comments
+    post.comments.order(updated_at: :desc).first(5)
+  end
+
+  private :update_posts_counter
```
Comment (Whoistolu) [inline app/models/post.rb:16]: - [x] [OPTIONAL] Great implementaton @fernando-silvabr66 , however, I would like ti to suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to the database, then make the update_post_counter method private.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :post
+
+  after_save :update_comments_counter
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
+
+  private :update_comments_counter
+end
```
Comment (Whoistolu) [inline app/models/comment.rb:12]: - [x] [OPTIONAL] Just as I have suggested earlier, I would like toto suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counter method every time a new post has been saved to the database, then make the update_comments_counter method private.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :post
+
+  after_save :update_likes_counter
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
+
+  private :update_likes_counter
+end
```
Comment (Whoistolu) [inline app/models/like.rb:12]: - [x] [OPTIONAL] Just as I have suggested earlier, I would like to suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_likes_counter method every time a new post has been saved to the database, then make the update_likes_counter method private.

**Review by Whoistolu — APPROVED** [review]
### Hi @fernando-silvabr66 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (fernando-silvabr66) [reply app/models/post.rb:16]: Thanks for the suggestion. I will study it a bit and try to implement in the next project's phase.

File: app/models/comment.rb (hunk 2 above)
Comment (fernando-silvabr66) [reply app/models/comment.rb:12]: Thanks for the suggestion. I will study it a bit and try to implement in the next project's phase.

File: app/models/like.rb (hunk 3 above)
Comment (fernando-silvabr66) [reply app/models/like.rb:12]: Thanks for the suggestion. I will study it a bit and try to implement in the next project's phase.

Comment (Whoistolu) [reply app/models/like.rb:12]: Great 👍🏽

**Pull request merged**